### DISCUSSION TITLE
Convert PWA app to Native apps - Ionic, Capacitor and Angular

### DISCUSSION THREAD

**Discussion by Sampath-Lokuge** [ionic-team/capacitor#3373] [Q&A] [answered]
Hi,

I have developed Ionic/Capacitor/Angular PWA app using this library `ng add @angular/pwa` and all are working fine. Now I need to use the same code base to build the Native apps too. So do I need to remove `Angular pwa` library and its configuration from my code to support native apps? Otherwise, will I have unexpected behavior on Native apps?

**Note:** Here I need both types of apps. i.e. PWA and Native apps

**Update** I have created 2 repos here. i.e. 2nd repo: removed Angular PWA library and all things related to PWA set up. But I do not think that is a good practice here since I need to update both projects with new enhancements. So what is the best architecture here to follow?

**Comment by dotNetkow** [comment, marked as answer]
Hi Sampath - you can use the same codebase to build the native apps. Here's an [example demo app](https://github.com/ionic-team/ionifits).

The main portions of a PWA are the app manifest and the service worker. App manifest is a configuration file - so that is ignored on mobile. Service Workers do not run in web views (which is how the app is displayed natively using Capacitor or Cordova) - so no effect should be seen.

**Reply by Sampath-Lokuge** [reply]
Hi,

Yes, you're right and I have used the same code base for both PWA and Native builds and no issues at all. Thanks a lot for the more info.

**Reply by dotNetkow** [reply]
Excellent! Glad to hear.
